encoding/encoding.go

```diff
@@ -43,33 +56,52 @@ func GetCompressor(name string) Compressor {
 	return registerCompressor[name]
 }
 
-type fancyCompressor struct {
-	name string
-	// cp has higher priority than Compressor set by WithCompressor
-	cp func(io.Writer) (io.WriteCloser, error)
-	// dc has lower priority than Compressor set by WithDecompressor
-	dc func(io.Reader) (io.Reader, error)
-}
-
-// NewFancyCompressor creates a Compressor which will be registered into a map.
-func NewFancyCompressor(cptype string,
-	cpFunc func(io.Writer) (io.WriteCloser, error),
-	dcFunc func(io.Reader) (io.Reader, error)) Compressor {
-	return &fancyCompressor{
-		name: cptype,
-		cp:   cpFunc,
-		dc:   dcFunc,
+// RegisterNewGIPCompressor registers gzip compressor.
+func RegisterNewGIPCompressor() {
```

Comment:
Please create a gzip subdirectory for a new gzip package and put this implementation in there.

(Also, it should register itself via an init().)